dataRetriever/resolvers/topicResolverSender/diffPeersListCreator.go

```diff
@@ -0,0 +1,84 @@
+package topicResolverSender
+
+import (
+	"bytes"
+
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+)
+
+// DiffPeersListCreator can create a peers list by making the set difference between peers on
+// main topic and the exclusion topic. If the resulting list is empty, will return the peers on the main topic.
+type DiffPeersListCreator struct {
+	messenger             dataRetriever.MessageHandler
+	mainTopic             string
+	excludePeersFromTopic string
+}
+
+// NewDiffPeersListCreator is the constructor for DiffPeersListCreator
+func NewDiffPeersListCreator(
+	messenger dataRetriever.MessageHandler,
+	mainTopic string,
```

Comment:
you could test to not have these 2 topics the same.

Reply:
I can do that as a slight optimization but the remaining code should be the same (maybe some one will use the struct with 2 different topics that hold the exactly the same same set of peers)

Reply:
This will actually make the code more messy as there is a situation when I call createShardHeaderResolver in metachain/resolversContainerFactory.go and this call will have to be guarded by an if as to not have the situation when the func is called with the same topic, example: shardHeadersForMetachain_1